gc/scheduler/scheduler.go

```diff
@@ -323,12 +331,20 @@ func (s *gcScheduler) run(ctx context.Context) {
 			}
 			s.waiters = nil
 			s.waiterL.Unlock()
+			gcRunSpan.RecordError(err)
```

Comment:
How about using `gcRunSpan.SetStatus(err)`? it records the error and also sets the span status to an Error code.